internal/validators/ip.go

```diff
@@ -0,0 +1,50 @@
+package validators
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+)
+
+var _ validator.String = IsIPNetwork{}
+
+// IsIPNetwork checks if a string is a valid IP network.
+type IsIPNetwork struct{}
+
+// Description describes the validation in plain text formatting.
+func (validator IsIPNetwork) Description(_ context.Context) string {
+	return "the string must be a valid IP network"
+}
+
+// MarkdownDescription describes the validation in Markdown formatting.
+func (validator IsIPNetwork) MarkdownDescription(ctx context.Context) string {
+	return validator.Description(ctx)
+}
+
+// Validate performs the validation.
+func (v IsIPNetwork) ValidateString(ctx context.Context, req validator.StringRequest, resp *validator.StringResponse) {
+	if req.ConfigValue.IsNull() || req.ConfigValue.IsUnknown() {
+		return
+	}
+
+	value := req.ConfigValue.ValueString()
+
+	ip := net.ParseIP(value)
+	if ip == nil { // Unable to parse the IP address
+
+		_, _, err := net.ParseCIDR(value)
+
+		if err != nil {
+			resp.Diagnostics.AddAttributeError(
+				req.Path,
+				"Invalid Network Value",
+				fmt.Sprintf("String must represent a valid IPv4 network, got: %s", value),
```

Comment:
The actual error message of the validation is probably too technical to add it to the diagnostics?

Reply:
> net.ParseCIDR("27.0.0.4/45")
> invalid CIDR address: 27.0.0.4/45

I can send the error directly, but not guarantee it will display the current value

Reply:
I send the error directly